cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/coprocessor/ConsumerConfigCache.java

```diff
@@ -232,12 +235,28 @@ public void run() {
     refreshThread.start();
   }
 
-  public static ConsumerConfigCache getInstance(Configuration hConf, byte[] tableName,
+  /**
+   * Sets an attribute to the given {@link Scan} object. Instead of calling {@link Scan#setAttribute(String, byte[])}
+   * directly, it uses reflection to call the method. This is because the return type for the setAttribute method
+   * is different in different HBase version.
+   */
+  private void setScanAttribute(Scan scan, String name, byte[] value) {
```

Comment:
Can't in here. This class is used inside Coprocessor, which doesn't have access to HBaseTableUtil. We can have the ScanBuilder being passed in through the getInstance method so that version specific coprocessor can pass in the right one. But seems like it just too much for a small operation like this.